lib/TransactionValidator.js

```diff
@@ -237,6 +242,120 @@ class TransactionValidator {
     return mismatches;
   }
 
+  getBodyMismatchObject(operationFromWSDL, error, isResponse, options, reason = '') {
+    const {
+        detailedBlobValidation
+      } = options,
+      mismatchReason = reason !== '' ? reason : (
+        detailedBlobValidation === true ?
+          error.message :
+          INVALID_BODY_DEFAULT_MESSAGE
+      );
+    let mismatch = {
+      property: isResponse ? RESPONSE_BODY_PROPERTY : BODY_PROPERTY,
+      transactionJsonPath: isResponse ? '$.response.body' : '$.request.body',
+      schemaJsonPath: operationFromWSDL.xpathInfo.xpath,
+      reasonCode: this.getMismatchReasonCode(error, detailedBlobValidation, isResponse),
+      reason: mismatchReason
+    };
+    return mismatch;
+  }
+
+  getMismatchWithSuggestedFix(mismatch, currentBody, bodyFromSchema, detailedBlobValidation) {
+    let newMismatch = JSON.parse(JSON.stringify(mismatch)),
+      key,
+      actualValue,
+      suggestedValue;
+    const {
+      reasonCode,
+      reason
+    } = newMismatch;
+
+    if (!detailedBlobValidation) {
+      key = 'body';
+      actualValue = currentBody;
+      suggestedValue = bodyFromSchema;
+    }
+    else {
+      const suggestedKey = (reasonCode === MISSING_IN_REQUEST_REASON_CODE) ?
+        this.getMissingChildKeyFromReason(reason) :
+        this.getKeyFromReason(reason);
+      key = this.getKeyFromReason(reason);
+      actualValue = this.getElementLineFromBody(currentBody, key);
+      suggestedValue = this.getElementLineFromBody(
+        bodyFromSchema,
+        suggestedKey
+      );
+    }
+    newMismatch.suggestedFix = {
+      key,
+      actualValue,
+      suggestedValue
+    };
+    return newMismatch;
+  }
+
+  getKeyFromReason(mismatchReason) {
+    const element = new RegExp(/Element '([^']+)':/),
+      quotedTextInElement = mismatchReason.match(element)[1];
+    return quotedTextInElement;
+  }
+
+  getMissingChildKeyFromReason(mismatchReason) {
+    const element = new RegExp(/Expected is \( ([A-Za-z]+) \)/),
+      quotedTextInElement = mismatchReason.match(element)[1];
+    return quotedTextInElement;
+  }
+
+  getElementLineFromBody(body, elementKey) {
+    const tagPattern = new RegExp(`<${elementKey}>[\\s\\S]*?<\/${elementKey}>|<${elementKey}[\\s\\S]*?\/>`),
+      matched = body.match(tagPattern);
+    return matched ? matched[0] : '';
```

Comment:
Are we handling when there is more than one property with the same name? are we going to check that with PM people?

Reply:
- Now we are handling when there are some elements with the same name in different levels (different parents for each one) with all type of error we support (invalid type, missing in the schema, missing in request, not allowed value).
- Now we are handling when there are some elements with the same name in the same level (all have the same parent), In this case, it is supported with some error types (invalid type, missing in the schema). some error types are not supported yet in this scenario (Missing in request and I did not test not allowed content yet).